### PR TITLE
[TRAFODION-2755] Create *Trafodion SQL Large Objects Guide* 2

### PR DESCRIPTION
**Pull request by liuyu000**

### CODE REVIEW

File: docs/lob_guide/src/asciidoc/_chapters/working_with_lob.adoc [hunk 1]
```diff
@@ -751,4 +751,380 @@ CQD `LOB_MAX_SIZE` (default 10G expressed in M [10000M]).
...
+
+Drop works like any other drop table. All dependent tables are deleted. All files in hdfs (data and descriptor) files are also deleted.
+
+For more information, see <<drop_table_statement,DROP TABLE Statement>> in http://trafodion.incubator.apache.org/docs/sql_reference/index.html[Trafodion SQL Reference Manual].
+
+[#garbage collection]
+== Garbage Collection
+
+When a lob datafile for a column has reached a certain limit, defined by a CQD `LOB_GC_LIMIT_SIZE`, then a compaction is triggered automatically. +
+The default GC Limit is 10GB and can be changed if needed.
```
Comment (DaveBirdsall) [inline docs/lob_guide/src/asciidoc/_chapters/working_with_lob.adoc]: Should define the GC acronym. For example, "The default Garbage Collection (GC) limit is ..." Then you can use GC later when you mean Garbage Collection.

File: docs/lob_guide/src/asciidoc/_chapters/working_with_lob.adoc [hunk 2]
```diff
@@ -751,4 +751,380 @@ CQD `LOB_MAX_SIZE` (default 10G expressed in M [10000M]).
...
+For more information, see <<drop_table_statement,DROP TABLE Statement>> in http://trafodion.incubator.apache.org/docs/sql_reference/index.html[Trafodion SQL Reference Manual].
+
+[#garbage collection]
+== Garbage Collection
+
+When a lob datafile for a column has reached a certain limit, defined by a CQD `LOB_GC_LIMIT_SIZE`, then a compaction is triggered automatically. +
+The default GC Limit is 10GB and can be changed if needed. 
+
+The need for GC arises because when a delete operation or an update operation is performed, the old data black in the hdfs file will be left as unused. +
+In the case of update, the old data will be left as unused and the new data will be written into a new section, so all these “holes” in the LOB data file are needlessly occupying space.
```
Comment (DaveBirdsall) [inline docs/lob_guide/src/asciidoc/_chapters/working_with_lob.adoc]: Consider: "... and the new data will be written into a new section, leaving "holes" in the LOB data file that needlessly take up space."

File: docs/lob_guide/src/asciidoc/_chapters/working_with_lob.adoc [hunk 3]
```diff
@@ -751,4 +751,380 @@ CQD `LOB_MAX_SIZE` (default 10G expressed in M [10000M]).
...
+
+The LOB descriptor chunks file is looked at to see which ranges and offsets are actually used. The LOB datafile is temporarily saved. The compaction is done into a new tempfile. When the sections have all been copied into the tempfile, Trafodion will delete the existing lob data file and rename the tempfile. 
+
+Finally, the saved copy of the LOB datafile is dropped. The saved copy is there just in case you need to fall back to it in case of an error. Since this operation is triggered as part of an IUD operation, a definite slowdown will occur for that insert/update operation compared to subsequent inserts/updates. 
+
+Also, each lob column of a table can be compacted separately as needed. GC does not have to be done to all columns of the LOB table all at once. 
+
+NOTE: Currently the GC is done in the same transaction as the transaction being used for the insert or update operation. If any part of the GC fails, then the entire transaction is aborted. 
+
+When Trafodion has support for local transactions, Trafodion will do the GC in a separate transaction or in a separate process, so you can fail the GC with a warning and allow the insert to go through.
```
Comment (DaveBirdsall) [inline docs/lob_guide/src/asciidoc/_chapters/working_with_lob.adoc]: Consider: "When Trafodion has support for local transactions, Trafodion will do the GC in a separate transaction or in a separate process, so GC can fail with a warning yet the insert can continue." (I'm assuming it is the system that causes the GC to fail and not the end-user. If I'm wrong, ignore this comment.)

File: docs/lob_guide/src/asciidoc/_chapters/working_with_lob.adoc [hunk 4]
```diff
@@ -751,4 +751,380 @@ CQD `LOB_MAX_SIZE` (default 10G expressed in M [10000M]).
...
+>>CREATE TABLE tlob130gt (c1 INT NOT NULL, c2 BLOB, c3 CLOB, c4 BLOB, PRIMARY KEY (c1));
+
+--- SQL operation complete.
+
+>>CREATE TABLE tlob130gt2 (c1 INT NOT NULL, c2 BLOB, c3 CLOB, c4 BLOB STORAGE 'EXTERNAL', PRIMARY KEY (c1));
+
+--- SQL operation complete.
+```
+
+* This Get Statement displays statistics for the table tlob130gt, the lob information is formatted for machine readability.
```
Comment (DaveBirdsall) [inline docs/lob_guide/src/asciidoc/_chapters/working_with_lob.adoc]: I'm confused. The example shows a SELECT statement, not a GET statement.

File: docs/lob_guide/src/asciidoc/_chapters/working_with_lob.adoc [hunk 5]
```diff
@@ -751,4 +751,380 @@ CQD `LOB_MAX_SIZE` (default 10G expressed in M [10000M]).
...
+ColumnName:  C4
+Lob Location:  External HDFS Location
+LOB Data File:  External HDFS File
+LOB EOD:  0
+LOB Used Len:  0
+
+--- SQL operation complete.
+```
+
+* This Get Statement displays statistics for the table tlob130gt2, the lob information is formatted for machine readability.
```
Comment (DaveBirdsall) [inline docs/lob_guide/src/asciidoc/_chapters/working_with_lob.adoc]: Select Statement?

File: docs/lob_guide/src/asciidoc/_chapters/working_with_lob.adoc (hunk 1 above)
Comment (liuyu000) [reply docs/lob_guide/src/asciidoc/_chapters/working_with_lob.adoc]: Thanks Dave.
I've corrected. :)

File: docs/lob_guide/src/asciidoc/_chapters/working_with_lob.adoc (hunk 2 above)
Comment (liuyu000) [reply docs/lob_guide/src/asciidoc/_chapters/working_with_lob.adoc]: Thanks Dave.
I've corrected. :)

File: docs/lob_guide/src/asciidoc/_chapters/working_with_lob.adoc (hunk 3 above)
Comment (liuyu000) [reply docs/lob_guide/src/asciidoc/_chapters/working_with_lob.adoc]: Thanks Dave.
I've corrected. :)

File: docs/lob_guide/src/asciidoc/_chapters/working_with_lob.adoc (hunk 4 above)
Comment (liuyu000) [reply docs/lob_guide/src/asciidoc/_chapters/working_with_lob.adoc]: Thanks Dave.
I've rewritten this part. :)

File: docs/lob_guide/src/asciidoc/_chapters/working_with_lob.adoc (hunk 5 above)
Comment (liuyu000) [reply docs/lob_guide/src/asciidoc/_chapters/working_with_lob.adoc]: Thanks Dave.
I've rewritten this part. :)

**Pull request merged**